GMG/Scenes/ChordProgress/ChordProgressView.swift

```diff
@@ -108,6 +114,93 @@ struct ChordProgressView: View {
 }
 
 extension ChordProgressView {
+    struct NavigationTitle: View {
+        @Environment(\.colorScheme) private var colorScheme: ColorScheme
+
+        @State var title: String
+        @State private var isTitleEditing = false
+        @FocusState private var isTitleFieldFocused: Bool
+
+        let onEnterTitle: (String) -> Void
+
+        init(
+            title: String,
+            onEnterTitle: @escaping (String) -> Void
+        ) {
+            self.title = title
+            self.onEnterTitle = onEnterTitle
+        }
+
+        private func toggleTitleEditing() {
+
+            if isTitleEditing {
+                finishEditingTitle()
+                return
+            }
+
+            isTitleEditing = true
+            DispatchQueue.main.async {
+                isTitleFieldFocused = true
+            }
+        }
```

Comment:
`startTitleEditing`과 `finishEditingTitle`로 명확히 구분하고 호출부에서 분기 처리하는 것이 좋을 것 같습니다.

```suggestion
        private func startTitleEditing() {
            isTitleEditing = true
            DispatchQueue.main.async {
                isTitleFieldFocused = true
            }
        }
```

Reply:
네 finishEditingTitle이 있으니, toggle보다는 start로 분리하는게 좋겠네요. 클린코드적으로도 맞는 방향인 것 같습니다! 감사합니다~!